lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -38,23 +38,33 @@ import type { PasswordlessSignInUpAction, SignInUpState, SignInUpChildProps, Nor
 import type { RecipeInterface } from "supertokens-web-js/recipe/passwordless";
 import type { User } from "supertokens-web-js/types";
 
-export const useSuccessInAnotherTabChecker = (
-    state: SignInUpState,
-    dispatch: React.Dispatch<PasswordlessSignInUpAction>,
-    userContext: any
-) => {
+export const useRedirectAfterSuccess = (state: SignInUpState, recipeId: string, userContext: any, history: any) => {
     const callingConsumeCodeRef = useRef(false);
 
     useEffect(() => {
         // We only need to start checking this if we have an active login attempt
-        if (state.loginAttemptInfo && !state.successInAnotherTab) {
+        if (state.loginAttemptInfo) {
             const checkSessionIntervalHandle = setInterval(async () => {
                 if (callingConsumeCodeRef.current === false) {
                     const hasSession = await Session.doesSessionExist({
                         userContext,
                     });
                     if (hasSession) {
-                        dispatch({ type: "successInAnotherTab" });
+                        clearInterval(checkSessionIntervalHandle);
+                        void SessionRecipe.getInstanceOrThrow().validateGlobalClaimsAndHandleSuccessRedirection(
+                            {
+                                rid: recipeId,
+                                successRedirectContext: {
+                                    action: "SUCCESS",
```

Comment:
Passing in `SUCCESS` as the action with `isNewRecipeUser: false` is not a good idea. Cause what if the user is actually a new signed up user? And based on that, the dev might want to take to the dashboard vs an onboarding screen? So in one tab, you would see an onboarding UI, and in another, you may see a dashboard UI (which might also break cause that UI might assume that onboarding is completed)

Since there is no other action available at the moment, I suggest that we remove this thing alltogether and just stick to showing the enter OTP screen / spinner screen - like how it's in other recipes.

Reply:
We decided to remove the `user` prop from here as we couldn't guarantee its presence.